pkg/iacinventory/github.go

```diff
@@ -0,0 +1,191 @@
+		if err != nil {
+			return nil, fmt.Errorf("error getting gh credentials from configuration files: %w", err)
+		}
+		if g.User == "" || g.OauthToken == "" {
+			return nil, fmt.Errorf("internal error: no credentials and no error during iacinventory.Run()")
+		}
+	}
+
+	// fetch the repositories associated with the account
+	if err := g.getRepos(); err != nil {
+		return nil, fmt.Errorf("error fetching github repositories: %w", err)
+	}
+
+	// print some nice, non-json info about notable repos
+	for _, repo := range g.repos {
+		ciMsg := "and has NO CI configuration."
+		if len(repo.getCISystems()) == 1 {
+			ciMsg = fmt.Sprintf("and is configured with %s CI.", repo.getCISystems()[0]) // TODO: include other CI systems
+		}
+		if len(repo.getCISystems()) > 1 {
+			ciMsg = "and is configured with multiple CI systems."
+		}
+		if len(repo.getTerraformDirs()) > 0 {
+			log.Infof("[%s]: contains %d Terraform directories %s\n", repo.Name, len(repo.getTerraformDirs()), ciMsg)
+		}
+		if len(repo.getCloudformationDirs()) > 0 {
+			log.Infof("[%s]: contains %d CloudFormation directories %s\n", repo.Name, len(repo.getCloudformationDirs()), ciMsg)
+		}
+	}
+
+	var out []Repo
+	for _, v := range g.repos {
+		out = append(out, v)
+	}
+	return out, nil
+}
+
+// githubCredsFromFS reads the `gh` configuration file on disk to get GitHub credentials.
+func githubCredsFromFS() (username string, oauthToken string, retErr error) {
+	var ghConfig map[string]interface{}
+	var ghConfigFile string
+
+	// Github credentials can exist in one of two files: ~/.config/gh/hosts.yml,
+	// or ~/.config/gh/config.yml. Creds _should_ only exist in the latter, but
+	// on some systems it appears in the former for whatever reason.
+	ghConfigFile, err := homedir.Expand("~/.config/gh/hosts.yml")
+	if err != nil {
+		retErr = fmt.Errorf("unable to get user Homedir: %w", err)
+		return
+	}
+	if _, err := os.Stat(ghConfigFile); err != nil {
+		if !os.IsNotExist(err) {
+			retErr = fmt.Errorf("unable to get gh config: %w", err)
+			return
+		}
+		// file does not exist, try another
+		ghConfigFile, err = homedir.Expand("~/.config/gh/config.yml")
+		if err != nil {
+			retErr = fmt.Errorf("unable to get user Homedir: %w", err)
+			return
+		}
+		if _, err := os.Stat(ghConfigFile); err != nil {
+			if !os.IsNotExist(err) {
+				retErr = fmt.Errorf("unable to get gh config: %w", err)
+				return
+			}
+			// still no file
+			retErr = fmt.Errorf("unable to locate GitHub `gh` configuration file")
+			return
+		}
+	}
+	yamlf, err := ioutil.ReadFile(ghConfigFile)
+	if err != nil {
+		retErr = fmt.Errorf("unable to read gh config: %w", err)
+		return
+	}
+	if err := yaml.Unmarshal(yamlf, &ghConfig); err != nil {
+		retErr = fmt.Errorf("unable to parse gh config: %w", err)
+		return
+	}
+
+	n := jnode.FromMap(ghConfig)
+	username = n.Path("github.com").Path("user").String()
+	oauthToken = n.Path("github.com").Path("oauth_token").String()
+	log.Infof(username)
```

Comment:
This function is currently broken, and I have no idea why. The empty-string checks below never trip, despite the credentials not being properly read-in. No idea what is going on here.

Reply:
String() returns JSON
GetString() returns a string value